### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by Kikobeats**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -21,114 +19,96 @@ class NanoClamp extends PureComponent {
...
+  componentWillUnmount () {
     window.removeEventListener('resize', this.debounced)
   }
 
-  debounce(func) {
+  debounce (func) {
     let timeout
 
     return () => {
-      const context = this
```
Comment (Kikobeats) [inline src/index.js:41]: removed because fat arrow bind the context

File: src/index.js [hunk 2]
```diff
@@ -21,114 +19,96 @@ class NanoClamp extends PureComponent {
...
       const later = () => {
         timeout = null
-        func.apply(context)
+        func.apply(this)
       }
       const callNow = !timeout
       clearTimeout(timeout)
-      timeout = setTimeout(later, 200)
```
Comment (Kikobeats) [inline src/index.js:48]: extract into a prop

File: src/index.js [hunk 3]
```diff
@@ -1,10 +1,8 @@
-import React, {createElement, PureComponent} from 'react'
+import {createElement, Component} from 'react'
 import PropTypes from 'prop-types'
 
-const ELLIPSIS = '...'
```
Comment (Kikobeats) [inline src/index.js:4]: extract into a prop

File: src/index.js [hunk 4]
```diff
@@ -21,114 +19,96 @@ class NanoClamp extends PureComponent {
...
 
-    return createElement(is, props, text)
+    return propText
+      ? createElement(is, { ref: node => (this.element = node), ...props }, text)
+      : null
   }
 }
 
 NanoClamp.defaultProps = {
-  className: '',
```
Comment (Kikobeats) [inline src/index.js:122]: Removed because now the component get all the `props`

File: src/index.js [hunk 5]
```diff
@@ -21,114 +19,96 @@ class NanoClamp extends PureComponent {
...
 }
 
 NanoClamp.propTypes = {
-  className: PropTypes.string,
   is: PropTypes.string,
   lines: PropTypes.number,
-  text: PropTypes.string.isRequired
+  debounce: PropTypes.number,
+  text: PropTypes.string.isRequired,
+  ellipsis: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
```
Comment (breadadams) [inline src/index.js]: What would be the need for `PropTypes.object`?

Comment (Kikobeats) [reply src/index.js]: true, not necessary

Comment (Kikobeats) [reply src/index.js]: for rendering a react component as ellipsis

**Pull request merged**